### PR TITLE
Add support for uri? in malli.transform (Clojure only)

### PR DESCRIPTION
**Pull request by stig**
With Reitit (0.7.0-alpha7), using the below route definition, I get a failure during the coercion when passing the uri "http://example.com".

    ["/uri"
      {:post {:summary "Post a URI - fails coercion"
              :coercion reitit.coercion.malli/coercion
              :parameters {:body {:uri uri?}}
              :responses {200 {:body {:uri uri?}}}
              :handler (fn [{{{uri :uri} :body} :parameters}]
                         {:status 200 :body {:uri uri}})}}]]

When I replaced the :coercion with the Clojure Spec version, the coercion works.

With this change the Malli coercer should also work.

I made it Clojure specific since it relies on `java.net.URI`.

### CODE REVIEW

File: src/malli/transform.cljc [hunk 1]
```diff
@@ -95,6 +96,15 @@
       x)
     x))
 
+#?(:clj
+   (defn -string->uri [x]
+     (if (string? x)
+       (try
+         (URI. x)
+         (catch Exception _
```
Comment (stig) [inline src/malli/transform.cljc:104]: Would it be better to catch `java.net.URISyntaxException` here?

Comment (opqdonut) [reply src/malli/transform.cljc:104]: Your call. The body of the `try` is so small there's little possibility of masking unrelated exceptions.

Comment (stig) [reply src/malli/transform.cljc]: You're right that the body is small, but I prefer `URISyntaxException` for explicitness - so I have updated it.

Comment (stig) [reply src/malli/transform.cljc]: Hmm, that fails the Babashka job, so I guess I change it back :-)

https://github.com/metosin/malli/actions/runs/6493910884/job/17635777011

Comment (stig) [reply src/malli/transform.cljc:106]: I couldn't get the recommended macro approach to work, so I just put a comment in to tighten up the catch clause once our version of babashka has the required commit.

**Comment by opqdonut** [comment]
good catch re: babashka!

**Pull request merged**